autoload/airline/extensions/branch.vim

```diff
@@ -109,6 +110,49 @@ function! s:get_hg_branch()
   return ''
 endfunction
 
+if s:has_async
+  let s:jobs = {}
```

Comment:
This dictionary is not cleared when a job is completed, this is a potential space leak if user opens a lot of file during an active session.

Please add a cleanup on job finish or job failure.